File: backend/Models/Plate.cs
using System.ComponentModel.DataAnnotations;

namespace PlateTracking.Models;

public class Plate
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(50)]
    public string SteelPlateNumber { get; set; } = string.Empty;

    [Required]
    public decimal DesignDepth { get; set; }

    [Required]
    public int LifeLimit { get; set; } = 10000;

    public int ImpressionCount { get; set; } = 0;

    public bool IsLocked { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Impression> Impressions { get; set; } = new List<Impression>();
    public ICollection<Warning> Warnings { get; set; } = new List<Warning>();
    public ICollection<AlignmentIncident> AlignmentIncidents { get; set; } = new List<AlignmentIncident>();
}

File: frontend/src/app/models/plate.ts
export interface PlateSummary {
  id: number;
  steelPlateNumber: string;
  designDepth: number;
  lifeLimit: number;
  impressionCount: number;
  isLocked: boolean;
  lifePercentage: number;
  createdAt: string;
}

export interface PlateDetail extends PlateSummary {
  impressions: Impression[];
  warnings: Warning[];
  incidents: Incident[];
}

export interface Impression {
  id: number;
  plateId: number;
  offsetX: number;
  offsetY: number;
  actualTemperature: number;
  createdAt: string;
}

export interface Warning {
  id: number;
  plateId: number;
  warningType: string;
  message: string;
  isAcknowledged: boolean;
  createdAt: string;
}

export interface Incident {
  id: number;
  plateId: number;
  steelPlateNumber: string;
  startImpressionId: number;
  endImpressionId: number;
  axis: string;
  notes?: string;
  isResolved: boolean;
  createdAt: string;
}

export interface CreatePlateRequest {
  steelPlateNumber: string;
  designDepth: number;
  lifeLimit: number;
}

export interface CreateImpressionRequest {
  plateId: number;
  offsetX: number;
  offsetY: number;
  actualTemperature: number;
}

export interface CreateImpressionResponse {
  impression: Impression;
  warnings: Warning[];
  incidents: Incident[];
  plateLocked: boolean;
}
